scripts/automated_ingestion/eessi_task.py

```diff
@@ -0,0 +1,1201 @@
+        return EESSITaskState.NEW_TASK
+
+    @log_function_entry_exit()
+    def _update_task_state_file(self, next_state: EESSITaskState, branch_name: str = None) -> Optional[Dict]:
+        """Update the TaskState file content in default or given branch"""
+        branch_name = self.git_repo.default_branch if branch_name is None else branch_name
+
+        task_pointer_file = self.description.task_object.remote_file_path
+        pull_request_dir = self._read_pull_request_dir_from_file(task_pointer_file, branch_name)
+        task_state_file_path = f"{pull_request_dir}/TaskState"
+        arch = self.description.get_metadata_filename_components()[3]
+        commit_message = f"change task state to {next_state} in {branch_name} for {arch}"
+        result = self._update_file(task_state_file_path,
+                                   f"{next_state.name}\n",
+                                   commit_message,
+                                   branch_name=branch_name)
+        return result
+
+    @log_function_entry_exit()
+    def _init_payload_object(self):
+        """Initialize the payload object"""
+        if self.payload is not None:
+            log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload object already initialized")
+            return
+
+        # get name of of payload from metadata
+        payload_name = self.description.metadata["payload"]["filename"]
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload_name: '%s'", payload_name)
+
+        # get config and remote_client from self.description.task_object
+        config = self.description.task_object.config
+        remote_client = self.description.task_object.remote_client
+
+        # determine remote_file_path by replacing basename of remote_file_path in self.description.task_object
+        #   with payload_name
+        description_remote_file_path = self.description.task_object.remote_file_path
+        payload_remote_file_path = os.path.join(os.path.dirname(description_remote_file_path), payload_name)
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "payload_remote_file_path: '%s'", payload_remote_file_path)
+
+        # initialize payload object
+        payload_object = EESSIDataAndSignatureObject(config, payload_remote_file_path, remote_client)
+        self.payload = EESSITaskPayload(payload_object)
+        log_message(LoggingScope.TASK_OPS, "INFO", "payload: '%s'", self.payload)
+
+    @log_function_entry_exit()
+    def _handle_add_new_task(self):
+        """Handler for ADD action in NEW_TASK state"""
+        log_message(LoggingScope.TASK_OPS, "INFO", "Handling ADD action in NEW_TASK state: '%s'",
+                    self.description.get_task_file_name())
+        # determine next state
+        next_state = self._next_state(EESSITaskState.NEW_TASK)
+        log_message(LoggingScope.TASK_OPS_DETAILS, "INFO", "next_state: '%s'", next_state)
+
+        # initialize payload object
+        self._init_payload_object()
```

Comment:
Maybe clarify a bit what this does (and maybe update the docstring for `_init_payload_object()` with a similar description. As far as I understand, it sets the paths to the remote and local file and signature locations, and then _also_ downloads the payload. That's relevant to know, because it makes it clear that this is a potentially time-consuming step if the payload is big.

In fact, maybe this should be in the description of this handler: in the "new task" state, the handler will download the payload, and progress to the 'payload_staged' state.